src/esockd_acceptor_sup.erl

```diff
@@ -55,8 +55,8 @@ count_acceptors(AcceptorSup) ->
 
 init([Proto, ListenOn, ConnSup, TuneFun, UpgradeFuns, Limiter]) ->
     SupFlags = #{strategy => simple_one_for_one,
-                 intensity => 100,
-                 period => 3600
+                 intensity => 100000,
```

Comment:
should be zero

Reply:
we need to consider this situation:
if the listener socket goes into a bad state, the acceptors will just keep looping restart with no possibility to recover.

maybe a better fix is to:
1. If acceptor encounters an error from the listener socket, it should shutdown with a loud error level log
2. The listener process should have a way to detect if the underlying socket is not healthy (if not restarted by acceptors).

Reply:
Need to review the sup tree design.

Reply:
> If acceptor encounters an error from the listener socket, it should shutdown with a loud error level log
added error logs.

>  The listener process should have a way to detect if the underlying socket is not healthy (if not restarted by acceptors due to reaching max restart intensity ).

Looks like it is ok to set intensity to 0 as the restarted listener adds new acceptors with newly opened listen socket.

This is confirmed in the newly added test case.